Products/ZenCollector/daemon.py

```diff
@@ -359,6 +370,25 @@ def generateEvent(self, event, **kw):
                 eventCopy['device_guid'] = guid
         return eventCopy
 
+    def should_trace_metric(self, metric, contextkey):
+        """
+        Tracer implementation - use this function to indicate whether a given
+        metric/contextkey combination is to be traced.
+        :param metric: name of the metric in question 
+        :param contextkey: context key of the metric in question
+        :return: boolean indicating whether to trace this metric/key
+        """
+        tests = []
+        if self.options.traceMetricName:
+            tests.append((self.options.traceMetricName, metric))
+        if self.options.traceMetricKey:
+            tests.append((self.options.traceMetricKey, contextkey))
+
+        result = [bool(re.compile(exp).match(subj)) for exp, subj in tests]
```

Comment:
I've taken out the compile() - per the note [here](https://docs.python.org/2/library/re.html#re.compile) and [this](https://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile) StackOverflow thread, search() and match() cache up to MAXCACHE (=100) expressions. So, this will compile the regex the first time it's used, and use the cached compiled version thereafter.